src/ProgramBlockEditor.js

```diff
@@ -248,11 +278,65 @@ class ProgramBlockEditor extends React.Component<ProgramBlockEditorProps, Progra
         this.insertSelectedCommandIntoProgram(stepNumber);
     };
 
+    // TODO: Discuss removing this once we have a good way to test drag and drop.
     /* istanbul ignore next */
-    handleDropCommand = (stepNumber: number) => {
+    handleDropCommandOnAddNode = (stepNumber: number) => {
+        // TODO: How to get to the event? Do we need this at all?
+        // event.preventDefault();
         this.insertSelectedCommandIntoProgram(stepNumber);
     };
 
+    // TODO: Discuss removing this once we have a good way to test drag and drop.
+    /* istanbul ignore next */
+    handleDragCommandOverProgramArea = (event: DragEvent) => {
+        event.preventDefault();
+
+        // Only attempt to recalculate the closest node every 100ms.
+        const timeStamp = Date.now();
+        if (timeStamp - this.lastCalculatedClosestAddNode > 100) {
+            const closestAddNodeIndex = this.findAddNodeClosestToEvent(event);
+            this.lastCalculatedClosestAddNode = timeStamp;
+
+            this.setState({
+                closestAddNodeIndex: closestAddNodeIndex
+            });
+        }
+    }
+
+    // TODO: Discuss removing this once we have a good way to test drag and drop.
+    /* istanbul ignore next */
+    handleDragLeaveOnProgramArea = (event: DragEvent) => {
+        const myBounds = this.programBlockEditorRef.current.getBoundingClientRect();
+        // Dragging over child elements with implicit drag and drop results in flickering.  Check to confirm whether we've actually left the area.
+        if (event.clientX >= myBounds.left &&
+            event.clientX <= (myBounds.left + myBounds.width) &&
+            event.clientY >= myBounds.top &&
+            event.clientY <= (myBounds.top + myBounds.height)) {
```

Comment:
If we invert the logic, we won't need the empty `if` block.

Reply:
Done.